test/GoodGhosting_No_Player_Wins.test.js

```diff
@@ -0,0 +1,119 @@
+                let playerEvent = "";
+                let paymentEvent = 0;
+                truffleAssert.eventEmitted(
+                    result,
+                    "JoinedGame",
+                    (ev) => {
+                        playerEvent = ev.player;
+                        paymentEvent = ev.amount;
+                        return playerEvent === player && new BN(paymentEvent).eq(new BN(segmentPayment));
+                    },
+                    `JoinedGame event should be emitted when an user joins the game with params\n
+                    player: expected ${player}; got ${playerEvent}\n
+                    paymentAmount: expected ${segmentPayment}; got ${paymentEvent}`,
+                );
+            }
+        });
+
+        it("runs the game - all players don not complete the game after joining", async () => {
+            // only depositing funds for the 1st payment window and running a loop to finish the game without any deposits
+            await timeMachine.advanceTime(segmentLength);
+            await goodGhosting.depositIntoExternalPool({ from: admin });
+            for (let segmentIndex = 2; segmentIndex < segmentCount; segmentIndex++) {
+                await timeMachine.advanceTime(segmentLength);
+            }
+            // the loop will run till segmentCount - 1
+            // completing the rest of the game
+            await timeMachine.advanceTime(segmentLength);
+            await truffleAssert.reverts(goodGhosting.depositIntoExternalPool({ from: admin }), "No amount from previous segment to deposit into protocol");
+            await timeMachine.advanceTime(segmentLength);
+        });
+
+
+        it("redeems funds from external pool", async () => {
+            let eventAmount = new BN(0);
+            const result = await goodGhosting.redeemFromExternalPool({ from: admin });
+            const contractsDaiBalance = new BN(await token.methods.balanceOf(goodGhosting.address).call({ from: admin }));
+            console.log("contractsDaiBalance", contractsDaiBalance.toString());
+            truffleAssert.eventEmitted(
+                result,
+                "FundsRedeemedFromExternalPool",
+                (ev) => {
+                    console.log("totalContractAmount", ev.totalAmount.toString());
+                    console.log("totalGamePrincipal", ev.totalGamePrincipal.toString());
+                    console.log("totalGameInterest", ev.totalGameInterest.toString());
+                    console.log("interestPerPlayer", ev.totalGameInterest.div(new BN(players.length - 1)).toString());
+                    // interest already transferred to the admin
+                    eventAmount = new BN(ev.totalGamePrincipal.toString());
+                    return eventAmount.eq(contractsDaiBalance);
+                },
+                `FundsRedeemedFromExternalPool error - event amount: ${eventAmount.toString()}; expectAmount: ${contractsDaiBalance.toString()}`,
+            );
```

Comment:
what do you think about confirming the admin balance to match the gameInterest?

Reply:
will add